src/utils.rs

```diff
@@ -10,6 +11,55 @@ pub unsafe fn debug_reset_static_data() {
     ffi::cairo_debug_reset_static_data()
 }
 
+pub fn status_to_result<T>(status: ffi::cairo_status_t, obj: T) -> Result<T, Error> {
+    match status {
+        ffi::STATUS_SUCCESS => Ok(obj),
+        ffi::STATUS_NO_MEMORY => Err(Error::NoMemory),
```

Comment:
Couldn't you use the `Into` impl on `Error` here?

Reply:
Indeed, done